mermithid/processors/misc/ComplexLineShape.py

```diff
@@ -0,0 +1,598 @@
+                strippeddirs = [s.strip('\n') for s in directory]
+                if 'scatter_spectra.npy' not in strippeddirs:
+                    self.generate_scatter_convolution_file()
+                test_file = self.path_to_osc_strengths_files+'scatter_spectra_file/scatter_spectra.npy' 
+                test_dict = np.load(test_file, allow_pickle = True)
+                if list(test_dict.item().keys())[0] != '{}_{}'.format(gases[0], gases[1]):
+                    print('first entry not matching, generating fresh files')
+                    self.generate_scatter_convolution_file()
+                elif len(test_dict.item()['{}_{}'.format(gases[0], gases[1])]) \
+                            != (1 + self.max_scatters+1)*(self.max_scatters+1)/2 - 1:
+                    print('Number of scatters not matching, generating fresh files')
+                elif len(test_dict.item()['{}_{}'.format(gases[0], gases[1])]['01_00'])\
+                != self.num_points_in_std_array:
+                    print('Binning do not match standard array, generating fresh files')
+                    self.generate_scatter_convolution_file()
+        return
+
+    # Given a function evaluated on the SELA, convolves it with a gaussian
+    def convolve_gaussian(self, func_to_convolve,gauss_FWHM_eV):
+        sigma = gaussian_FWHM_to_sigma(gauss_FWHM_eV)
+        resolution_f = self.std_gaussian(sigma)
+        ans = sp.signal.convolve(resolution_f,func_to_convolve,mode='same')
+        ans_normed = self.normalize(ans)
+        return ans_normed
+
+
+    def make_spectrum(self, gauss_FWHM_eV, prob_parameter, scatter_proportion, emitted_peak='shake'):
+        gases = self.gases
+        max_scatters = self.max_scatters
+        max_comprehensive_scatters = self.max_comprehensive_scatters
+        current_path = self.path_to_osc_strengths_files
+        # check_existence_of_scatter_files()
+        #filenames = list_files('scatter_spectra_files')
+        p = scatter_proportion
+        q = 1 - p
+        scatter_spectra = np.load(
+        current_path + 'scatter_spectra_file/scatter_spectra.npy', allow_pickle = True
+        )
+        en_array = self.std_eV_array()
+        current_full_spectrum = np.zeros(len(en_array))
+        if emitted_peak == 'lorentzian':
+            current_working_spectrum = self.std_lorenztian_17keV()
+        elif emitted_peak == 'shake':
+            current_working_spectrum = self.shakeSpectrumClassInstance.shake_spectrum()
+        current_working_spectrum = self.convolve_gaussian(current_working_spectrum, gauss_FWHM_eV)
+        zeroth_order_peak = current_working_spectrum
+        current_full_spectrum += current_working_spectrum
+        for n in range(1, max_comprehensive_scatters + 1):
+            for r in range(0, n + 1):
+                current_working_spectrum = \
+                scatter_spectra.item()['{}_{}'.format(gases[0], gases[1])]\
```

Comment:
one of the nice features about the previous implementation of the line fitting model, was that you could either do both gases, or just one of the gases easily, or extend the model to N gases relatively easy, if you had the data file

Will this code break if nGases!=2? it seems like it to me... please correct me if Im wrong